src/main/java/com/thealgorithms/maths/SecondMinMax.java

```diff
@@ -0,0 +1,48 @@
+package com.thealgorithms.maths;
+
+public final class SecondMinMax {
+
+    /**
+     * @brief Finds the Second minimum / maximum value from the array
+     * @param int array
+     * @exception IllegalArgumentException if input array is empty or length 1 also if all elements are same
+     * @return the second minimum / maximum value from the input array
+     * @author Bharath Sanjeevi ( https://github.com/BharathSanjeeviT )
+     */
+
+    private SecondMinMax() {
+    }
+
+    public static int findSecondMin(final int[] arr) {
+        int secMin = Integer.MAX_VALUE, min = Integer.MAX_VALUE;
+        checkInput(arr);
+        for (final int num : arr)
+            if (num < min) {
+                secMin = min;
+                min = num;
+            } else if ((num < secMin) && (num != min))
+                secMin = num;
+        if (secMin == Integer.MAX_VALUE) throw new IllegalArgumentException("Cant find Second Maximum / Minimum in array full of same numbers");
```

Comment:
What do you think about introducing a method `checkOutput(...)`? I thnk it it enough to check if _best_ and _secondBest_ are the same (here `min` and `secondMin`).

Reply:
Yeah, this change looks good 👍